xbmc/utils/CharsetConverter.cpp

```diff
@@ -495,7 +499,8 @@ bool CCharsetConverter::CInnerConverter::logicalToVisualBiDi(const std::u32strin
 
     bool bidiFailed = false;
     FriBidiCharType baseCopy = base; // preserve same value for all lines, required because fribidi_log2vis will modify parameter value
-    if (fribidi_log2vis((const FriBidiChar*)(stringSrc.c_str() + lineStart), lineLen, &baseCopy, visual, NULL, NULL, NULL))
+    if (fribidi_log2vis((const FriBidiChar*)(stringSrc.c_str() + lineStart), lineLen, &baseCopy,
+      visual, logicalToVisualMap, NULL, NULL))
```

Comment:
Align to other arguments, space on cast and nullptr.

```cpp
if (fribidi_log2vis((const FriBidiChar*) (stringSrc.c_str() + lineStart),
                    lineLen, &baseCopy, visual, logicalToVisualMap, nullptr, nullptr))
```

I have no idea what the correct c++ cast is there. Maybe `reinterpret_cast`? Try it and see.